libs/utils/android/workloads/camerarecord.py

```diff
@@ -0,0 +1,112 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2015, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from android import Wlauto
+from android import Workload
+
+import os
+import glob
+import logging
+
+from shutil import copyfile, rmtree
+
+class CameraRecord(Wlauto):
+    """
+    Camera Record workload
+    """
+
+    # Package required by this workload
+    package = 'com.google.android.GoogleCamera'
+
+    # WA name of the workload
+    workload_name = 'camerarecord'
+
+    # Setup logger
+    logger = logging.getLogger('Camera Record')
+    logger.setLevel(logging.INFO)
+
+    def __init__(self, test_env):
+        super(CameraRecord, self).__init__(test_env)
+        logging.debug('%15s - Workload created', CameraRecord.workload_name)
+
+    def run(self, exp_dir, agenda, collect=''):
+        """
+        :param exp_dir: Path to experiment directory where to store results.
+        :type exp_dir: str
+
+        :param agenda: Agenda to be passed to workload-automation. Can be a
+            YAML file or a dictionary.
+        :type agenda: str or dict
+
+        :param collect: Specifies what to collect. Possible values:
+            - 'energy'
+        :type collect: list(str)
+        """
+
+        # Initialize energy meter results
+        nrg_data, nrg_file = None, None
+
+        # Setting agenda_yaml properly
+        if type(agenda) is str:
+            agenda_yaml = agenda
+        else:
+            # Convert dictionary to YAML file
+            agenda_yaml = self.generate_yaml_agenda(agenda)
+
+        logging.debug('%14s - Running', CameraRecord.workload_name)
+
+        # Start energy collection
+        if 'energy' in collect and self.te.emeter:
+            self.te.emeter.reset()
+
+        wa_out_dir = Wlauto.wa_run(agenda_yaml)
+
+        # Stop energy collection
+        if 'energy' in collect and self.te.emeter:
+            nrg_data, nrg_file = self.te.emeter.report(exp_dir)
+            logging.info("Estimated energy: %7.3f", float(nrg_data['BAT']))
```

Comment:
Here we still have that hardcoded BAT lable... have we never decided how to make that more generic?